src/components/card/card.tsx

```diff
@@ -87,17 +96,37 @@ export class Card {
     @Event()
     public actionSelected: EventEmitter<ActionBarItem>;
 
+    @Element()
+    private host: HTMLElement;
+
+    private handleMouseEnter: () => void;
+    private handleMouseLeave: () => void;
+
+    public componentWillLoad() {
+        const { handleMouseEnter, handleMouseLeave } = getMouseEventHandlers(
+            this.host,
+        );
+        this.handleMouseEnter = handleMouseEnter;
+        this.handleMouseLeave = handleMouseLeave;
+    }
+
     public render() {
         return (
-            <section tabindex={this.clickable ? 0 : ''}>
-                {this.renderImage()}
-                <div class="body">
-                    {this.renderHeader()}
-                    {this.renderSlot()}
-                    {this.renderValue()}
-                    {this.renderActionBar()}
-                </div>
-            </section>
+            <Host
+                onMouseEnter={this.handleMouseEnter}
+                onMouseLeave={this.handleMouseLeave}
+            >
+                <section tabindex={this.clickable ? 0 : ''}>
+                    {this.renderImage()}
+                    <div class="body">
+                        {this.renderHeader()}
+                        {this.renderSlot()}
+                        {this.renderValue()}
+                        {this.renderActionBar()}
+                    </div>
+                    <div class="limel-3d-hover-effect-glow" />
```

Comment:
I think there are more things I can clean up, and built into the helper-function. Like this element for example.

Reply:
I'm sure you can make this way better. Especially, if we can export this helper function to our other repos, it would sound like a good idea. But we can always improve these things further, by creating new issues; if moving forwards is more important at the moment. 

This is an internal feature right now, and styling the "glow" div (inside the consumer component) needs some CSS considerations as well. We can even turn this div into a `private` component which can carry its own styles, to avoid unnecessary documentations and usage of mixins, etc… and also to add further flexibility using props, e.g. allowing consumer to control the intensity or color of the glow, etc…

I think all of these can be done later though. At the moment, I'm not sure if we will use this effect in many other components. This effect is only useful in very particular UI components.

Reply:
@adrianschmidt If we wanted to make `limel-3d-hover-effect-glow` its own `@private` component, where would be the correct place to add such a component? In the `util` folder? And what would be a good name for it?

Reply:
All components, private or not, should go in their own folder in the `components` folder, according to Stencil's style guide. We've departed from that in some cases, but I think we can put it there. And `limel-3d-hover-effect-glow` seems like a good name to me 👍

It would be great if we could contain the styling in a component! Or at least part of the styling.